tests/units/DB.php

```diff
@@ -913,4 +913,323 @@ function () {
+                    $positive_integer_values,
+                    $negative_integer_values,
+                    $float_values,
+                ),
+                'invalid_values' => array_merge(
+                    $text_values,
+                    $date_values,
+                    $time_values,
+                    $datetime_values,
+                ),
+            ];
+        }
+
+        // date
+        yield [
+            'table'  => 'glpi_budgets',
+            'field'  => 'begin_date',
+            'pattern' => '/^(NULL|null|(?:\^|(>|<|>=|<=)\s*|%)?[\d\-]+(%|\$)?)$/',
+            'valid_values' => array_merge(
+                $null_values,
+                $date_values,
+                $relative_datetime_values,
+                [
+                    // cannot be put in `$date_values` as it would not produce expected result for integer/float fields
+                    '2023',
+                ],
+                // integer values are accepted, as it correspond to $relative_datetime_values pattern
+                $positive_integer_values,
+                $negative_integer_values,
+            ),
+            'invalid_values' => array_merge(
+                $text_values,
+                $float_values,
+                $time_values,
+                $datetime_values,
+            ),
+        ];
+
+        // timestamp
+        yield [
+            'table'  => 'glpi_tickets',
+            'field'  => 'solvedate',
+            'pattern' => '/^(NULL|null|(?:\^|(>|<|>=|<=)\s*|%)?[\d:\- ]+(%|\$)?)$/',
+            'valid_values' => array_merge(
+                $null_values,
+                $date_values,
+                $time_values,
+                $datetime_values,
+                $relative_datetime_values,
+                // integer values are accepted, as it correspond to $relative_datetime_values pattern
+                $positive_integer_values,
+                $negative_integer_values,
+            ),
+            'invalid_values' => array_merge(
+                $text_values,
+                $float_values,
+            ),
+        ];
+
+        // time
+        /*
+         * There is actually no `time` field that is searchable.
+         * Should we support this?
+        yield [
+            'table'  => 'glpi_calendarsegments',
+            'field'  => 'begin',
+            'pattern' => '/^(NULL|null|(?:\^|(>|<|>=|<=)\s*|%)?[\d:]+(%|\$)?)$/',
+            'valid_values' => array_merge(
+                $null_values,
+                $time_values,
+                // integer values are accepted, as it correspond to $relative_datetime_values pattern
+                $positive_integer_values,
+                $negative_integer_values,
+            ),
+            'invalid_values' => array_merge(
+                $text_values,
+                $date_values,
+                $float_values,
+                $datetime_values,
+                $relative_datetime_values,
+            ),
+        ];
+        */
```

Comment:
Tests were failing here due to some unexpected values allowed by the pattern, but I figured out that there is no `time` field that can be used in search, so I think we should just drop this case.